cmd/lncli/commands.go

```diff
@@ -2080,6 +2080,20 @@ func parseChanPoint(s string) (*lnrpc.ChannelPoint, error) {
 	}, nil
 }
 
+// parseTimeLockDelta is expected to get a uint16 type of timeLockDelta,
+// which maximum value is MaxTimeLockDelta.
+func parseTimeLockDelta(timeLockDeltaStr string) (uint16, error) {
+	timeLockDeltaUnCheck, err := strconv.ParseUint(
+		timeLockDeltaStr, 10, 64,
+	)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse time_lock_delta: %s "+
+			"to uint64, err: %v", timeLockDeltaStr, err)
+	}
+
+	return uint16(timeLockDeltaUnCheck), nil
```

Comment:
I think we should check that the value isn't greater than `uint64(routing.MaxCLTVDelta)` before we cast to `uint16`. Otherwise a value greater than `uint16` won't result in an error but a possibly unexpected high/truncated `uint16` value.

Reply:
I followed @ellemouton 's review comment: https://github.com/lightningnetwork/lnd/pull/7350#discussion_r1090295163

How about add a condition check : `if timeLockDeltaUnCheck > math.MaxUint16 {` here?

Reply:
Yes, that comment wasn't wrong per se. But the cast to `uint16` actually suppresses any errors on the server side, as the value can never be larger than the maximum. This would lead to silent but unexpected behavior.

The current version looks better. But given we now have two constants for this, we should use `routing.MaxCLTVDelata` here instead of `math.MaxUint16`.

Reply:
Agree, I will update this to use `routing.MaxCLTVDelata`

Reply:
Updated.